### DISCUSSION TITLE
Add Thrift Support

### DISCUSSION THREAD

**Discussion by Kaitachi** [Kong/insomnia#5469] [Ideas]
Thrift Services are an alternative to gRPC, although with some nuances. It is developed by [Apache](https://thrift.apache.org).

Can't say much about Thrift other than the fact that I'm using it at my workplace to communicate across various services and that it would be useful to have a convenient and straightforward way to build Thrift requests with ease, similar to what is currently available with gRPC.

Sure there exist ways to assemble Thrift requests as REST requests, but it'd be nice to have all the bells and whistles of having a predefined service file and just upload it to Insomnia to use it right away.

Would there be any objections or considerations to be made before moving forward with this idea?

I'm willing to move forward and begin development of this, but I'd also like to generate some discussion and gather feedback about this suggestion. I'm relatively new in the open-source contribution ecosystem and looking forward to making significant contributions.

Thoughts? Suggestions?

**Comment by isnikulin** [comment]
Very surprised there doesn't exist any support for thrift services already.
